### PR TITLE
adding agents intro files

### PR DESCRIPTION
**Pull request by rsgrewal-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb [hunk 1]
```diff
@@ -0,0 +1,1294 @@
...
+    "1. The user enters an input to the application\n",
+    "2. The user input is merged with the original prompt and sent to the LLM to determine the next step\n",
+    "3. If the LLM knows the answer, it will answer and we are done. If not, go to next step 4.\n",
+    "4. The LLM will determine which tool to use to answer the question.\n",
+    "5. We will use the tool as directed by the LLM and retrieve the results.\n",
+    "6. We provide the results back into the original prompt as more context.\n",
+    "7. We ask the LLM the next step or if knows the answer.\n",
+    "8. Return to step 3.\n",
+    "\n",
+    "If this is a bit confusing do not panic, we will walk through this flow in an example shortly!"
```
Comment (mani-aiml) [inline 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb:266]: lets not use the word 'panic'.

File: 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb [hunk 2]
```diff
@@ -0,0 +1,1294 @@
...
+    "    response = requests.get(url, params=params).json()\n",
+    "\n",
+    "    if response:\n",
+    "        lat = response[0][\"lat\"]\n",
+    "        lon = response[0][\"lon\"]\n",
+    "        return {\"latitude\": lat, \"longitude\": lon}\n",
+    "    else:\n",
+    "        return None\n",
+    "\n",
+    "@tool (\"get_weather_dict\")\n",
```
Comment (mani-aiml) [inline 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb:488]: Can we annotate the previous function or do we need to redefine the function for using with Langchain?

**Review by mani-aiml — CHANGES_REQUESTED** [review]

File: 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb [hunk 3]
```diff
@@ -0,0 +1,1294 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "88a5ab2f-d044-4956-b75b-7408d9c3e323",
+   "metadata": {},
+   "source": [
+    "# Invoking a function or do-it-yourself Agents with Bedrock\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "Conversational interfaces such as chatbots and virtual assistants can be used to enhance the user experience for your customers. These use natural language processing (NLP) and machine learning algorithms to understand and respond to user queries and can be used in a variety of applications, such as customer service, sales, and e-commerce, to provide quick and efficient responses to users. usuallythey are augmented by fetching information from various channels such as websites, social media platforms, and messaging apps which involve a complex workflow as shown below\n",
```
[the remark this answers is not in the document]
Comment (rsgrewal-aws) [reply 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb]: fixed in the latest push

File: 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb [hunk 4]
```diff
@@ -0,0 +1,1294 @@
...
+    "In Conversational interfaces such as chatbots, it is highly important to remember previous interactions, both at a short term but also at a long term level. Further we need to enhance with external tools which are needed to complete the queries. There are many providers which help to create functions or agents. We will look at a vanila implementation for how to do it yourself\n",
+    "\n",
+    "### Building  - Key Elements\n",
+    "\n",
+    "The first process in a building a contextual-aware chatbot is to identify the tools which can be called by the LLM's.\n",
+    "\n",
+    "Second process is the user request orchestration , interaction,  invoking and returing the results\n",
+    "\n",
+    "### Architecture [Weather lookup]\n",
+    "We Search and look for the Latitude and Longitude and then invoke the weather app to get predictions\n",
```
[the remark this answers is not in the document]
Comment (rsgrewal-aws) [reply 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb:37]: Good catch, have added some additional info and context for this information

File: 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb [hunk 5]
```diff
@@ -0,0 +1,1294 @@
...
+    "print(f'Weather in {place} is...')\n",
+    "weather_response"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "18203223",
+   "metadata": {},
+   "source": [
+    "As you might expect, we have to describe our tools to our LLM, so it knows how to use them. The strings below describe the python functions for lat/long and weather to Claude in an XML friendly format which we have seen previously in the workshop."
```
[the remark this answers is not in the document]
Comment (rsgrewal-aws) [reply 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb]: fixed in the latest push

File: 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb (hunk 1 above)
Comment (rsgrewal-aws) [reply 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb]: thanks and good catch, have adjusted the wordings

File: 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb (hunk 2 above)
Comment (rsgrewal-aws) [reply 07_Agents/weather_service_agent/00_Function_DIY_Agents.ipynb:490]: we need to annotate separately

**Comment by rsgrewal-aws** [comment]
added the changes in the latest push

**Review by mani-aiml — APPROVED** [review]
Approved with latest changes.

**Pull request merged**